src/persist/src/operators/stream.rs

```diff
@@ -265,58 +264,106 @@ where
-                    // the case, but antichains of types with a different partial order
-                    // can have frontier progress and have some elements that don't
-                    // represent progress.
-                    if !input_frontier.less_than(frontier_element) {
-                        continue;
-                    }
-
-                    log::trace!("Sealing {} up to {}", &operator_name, frontier_element);
-
-                    if let Err(e) = write.seal(*frontier_element).await {
-                        // If we fail to seal, simply exit and try again the next
-                        // time the operator is scheduled.
-                        log::error!(
-                            "Error sealing {} up to {}: {:?}",
+                    if input_frontier.less_than(&frontier_element) {
+                        log::trace!(
+                            "In {}, sealing collection up to {}...",
                             &operator_name,
                             frontier_element,
-                            e
                         );
 
-                        // Reschedule this operator after a small delay to retry the
-                        // seal.
-                        activator.activate_after(Duration::from_millis(100));
+                        let future = write.seal(*frontier_element);
 
-                        return;
+                        pending_futures.push_back(SealFuture {
+                            time: *frontier_element,
+                            future,
+                        });
                     }
                 }
 
-                input_frontier = new_input_frontier.to_owned();
-                if input_frontier.is_empty() {
-                    initial_capabilities.clear();
-                }
+                // Swing through all pending futures and see if they're ready. Ready futures will
+                // invoke the Activator, which will make sure that we arrive here, even when there
+                // are no changes in the input frontier or new input.
+                let waker = futures_util::task::waker_ref(&activator);
+                let mut context = Context::from_waker(&waker);
+
+                while let Some(mut pending_future) = pending_futures.pop_front() {
+                    match Pin::new(&mut pending_future.future).poll(&mut context) {
+                        std::task::Poll::Ready(Ok(_)) => {
+                            log::trace!(
+                                "In {}, finished sealing collection up to {}",
+                                &operator_name,
+                                pending_future.time,
+                            );
+                            // Explicitly downgrade the capability to the new time.
+                            cap_set.downgrade(Some(pending_future.time));
+                        }
+                        std::task::Poll::Ready(Err(e)) => {
+                            let error = format!(
```

Comment:
nit: This is a remnant from my code, when it was still both emitting the error and logging it. Now you could just put this directly in the `log::trace!()` call.